FeedSX/screens/createPost/index.tsx

```diff
@@ -0,0 +1,549 @@
+
+  // convert the default selected image/video format to LMAttachmentUI
+  function convertData(data: imageDefaultFormat[]): LMAttachmentUI[] {
+    const convertedData = data.map(item => {
+      return {
+        attachmentMeta: {
+          entityId: '', // You need to specify the entity ID.
+          format: item.type,
+          name: item.fileName,
+          ogTags: {
+            description: '',
+            title: '',
+            url: '',
+            image: '',
+          },
+          size: item.fileSize,
+          duration: item.duration, // You need to specify the duration.
+          pageCount: 0, // You need to specify the page count.
+          url: item.uri,
+        },
+        attachmentType:
+          item.type === 'image/jpg'
+            ? IMAGE_ATTACHMENT_TYPE
+            : item.type === 'video/mp4'
+            ? VIDEO_ATTACHMENT_TYPE
+            : 0, // You need to specify the attachment type.
+      };
+    });
+    setFormattedMediaAttachments([
+      ...formattedMediaAttachments,
+      ...convertedData,
+    ]);
+    return convertedData;
+  }
+
+  // convert the default selected document format to LMAttachmentUI
+  function convertDocData(data: docDefaultFormat[]): LMAttachmentUI[] {
+    const convertedData = data.map(item => {
+      return {
+        attachmentMeta: {
+          entityId: '', // You need to specify the entity ID.
+          format: item.type,
+          name: item.name,
+          ogTags: {
+            description: '',
+            title: '',
+            url: '',
+            image: '',
+          },
+          size: item.size,
+          duration: 0, // You need to specify the duration.
+          pageCount: 0, // You need to specify the page count.
+          url: item.uri,
+        },
+        attachmentType:
+          item.type === 'application/pdf' ? DOCUMENT_ATTACHMENT_TYPE : 0, // You need to specify the attachment type.
+      };
+    });
+    setFormattedDocAttachments([...formattedDocAttachments, ...convertedData]);
+    return convertedData;
+  }
+
+  // convert the LMOGTagsUI to LMAttachmentUI
+  function convertLinkData(data: LMOGTagsUI[]): LMAttachmentUI[] {
+    const convertedData = data.map(item => {
+      return {
+        attachmentMeta: {
+          entityId: '', // You need to specify the entity ID.
+          format: '',
+          name: '',
+          ogTags: {
+            description: item.description,
+            title: item.title,
+            url: item.url,
+            image: item.image,
+          },
+          size: 0,
+          duration: 0, // You need to specify the duration.
+          pageCount: 0, // You need to specify the page count.
+          url: '',
+        },
+        attachmentType: LINK_ATTACHMENT_TYPE, // You need to specify the attachment type.
+      };
+    });
+    setFormattedLinkAttachments([
+      ...formattedLinkAttachments,
+      ...convertedData,
+    ]);
+    return convertedData;
+  }
```

Comment:
move this convertor to a new file and then use returned data in state. and rename `convertLinkMetaData`

Reply:
resolved